server/log_rpc_server.go

```diff
@@ -812,3 +812,13 @@ func (t *TrillianLogRPCServer) recordIndexPercent(leafIndex int64, treeSize uint
 func spanFor(ctx context.Context, name string) (context.Context, *trace.Span) {
 	return trace.StartSpan(ctx, fmt.Sprintf("%s.%s", traceSpanRoot, name))
 }
+
+func (t *TrillianLogRPCServer) snapshotForTree(ctx context.Context, tree *trillian.Tree, method string) (storage.ReadOnlyLogTreeTX, error) {
+	tx, err := t.registry.LogStorage.SnapshotForTree(ctx, tree)
+	if err == storage.ErrTreeNeedsInit {
```

Comment:
For consistency, would it be better to have this close whenever `err != nil && tx != nil`?

Reply:
Could do if you think it helps. It's already closed though except for when this error happens. I wish we'd never added the special case but too late now.

Reply:
Done.